### DISCUSSION TITLE
Keep getting "path not allowed on the configured scope" with `fs.readTextFile`

### DISCUSSION THREAD

**Discussion by taviks** [tauri-apps/tauri#5990] [Q&A]
For the life of me I can't get the below to work:

 ```js
const fileContent = await tauri.fs.readTextFile(
    "path-relative-to-home-dir/test-file.json5", 
    { dir: tauri.fs.BaseDirectory.Home }
).then(console.log)
```

Keep getting: 

> Unhandled Promise Rejection: path not allowed on the configured scope {redacted-path-here}

The relevant section of my Tauri config file:

```json5
tauri: {
   allowlist: {
      "fs": {
         all: true, // enable all FS APIs
         scope: ["**"]
         ...
```

For the `scope` property I've tried lots of different thing:

```
["**"]
["**/*"]
["$HOME"]
["$HOME/**"]
["$HOME/**/*"]
```

None of these work. 

The _only_ thing I can get to work is using the full path, e.g. `["$HOME/path-relative-to-home-dir/test-file.json5"]`

---

**Version details:**

Tauri: 1.2.1
@tauri-apps/api: 1.2.0
@tauri-apps/cli: 1.2.2
Rust: 1.65
Node: 16.17.0

---

Any ideas?

**Comment by deadlinecode** [comment]
```json
{
  "scope": ["**", "**/*", "/**/*"]
}
```
Worked for me
Problem was that normally tauri doesn't allow absolut paths

**Reply by FabianLars** [reply]
Just for anyone else reading this, we generally do not recommend scopes like this at all, the scope should always be as restrictive as possible.

I'm not sure what OP's issue is/was, To my knowledge there's no general bug here 🤔 

Also in case pre-defiend scopes don't work for every use case, tauri's dialog and file drop mechanisms will dynamically extend the scope, and you can extend the scope at runtime in rust too: https://docs.rs/tauri/latest/tauri/scope/struct.FsScope.html (example: `app.fs_scope().allow_file(path);`)

**Reply by SanGuiChen** [reply]
Can you give me an example if it's convenient?

**Reply by FabianLars** [reply]
An example for what?

**Reply by SanGuiChen** [reply]
I hope I can dynamically modify the scopes of fs in the rust section. Since I am a novice in rust, I would appreciate it if you could give me an example of modifying scope.

**Reply by FabianLars** [reply]
There honestly isn't much more to it then [this](https://github.com/tauri-apps/tauri/discussions/5990#discussioncomment-5633881). What about a tauri::command specific example then:
```rs
#[tauri::command]
fn expand_scope(app_handle: tauri::AppHandle, folder_path: std::path::PathBuf) -> Result<(), String> {
  // If possible, verify your path if it comes from your frontend.

  // true means that we want inner directories allowed too
  app_handle.fs_scope().allow_directory(&folder_path, true)
    .map_err(|err| err.to_string())
}
```
then invoke it from js like this
```ts
invoke("expand_scope", { folderPath: somePathWhatever })
```

Note, i generally would not recommend this approach. It's intentional that you can't extend the scope at runtime for security reasons. It's just the easiest way to showcase it.

**Reply by SanGuiChen** [reply]
I'll see if this method works. Thank you very much!

**Comment by jamischarles** [comment]
For Tauri 2 this worked great, but I had to include this first:
```rs
use tauri_plugin_fs::FsExt;
```
